### PR TITLE
DevTest: `chest_of_everything` mod: Add search, bag, improve formspec

### LINKED ISSUE #12825: Devtest chest of everything needs a search field
## Problem

Finding the item I want in the devtest chest of everything is annoying and takes too much time.

## Solutions

Add a search field, like in mtg creative.

## Alternatives

Manually group items by their functionality.

## Additional context

No.

### PR DESCRIPTION
**Pull request by Wuzzy2**
I've backported my `getitem` mod into DevTest's `chest_of_everything` mod. These are the improvements/changes:

* Added Bag of Everything (IMHO more convenient to use than the Chest of Everything)
* `give_initial_items` mod gives you a Bag of Everything instead of a Chest of Everything
* Added search feature (fixes #12825)
* Increased number of slots per page to 5×10 (was: 4×8)
* Detached inventories are now properly managed per-player instead of globally

This should greatly improve usability overall.

DevTest-specific features (such as the DevTest-specific sorting of items, or the lack of privilege checks) are kept. The mod name `chest_of_everything` is kept as this mod is not identical to `getitem` and shouldn't be.

## To do

This PR is ready for review.

## How to test

* In DevTest, use the item `chest_of_everything:chest` or `chest_of_everything:bag`.

### CODE REVIEW

File: games/devtest/mods/chest_of_everything/init.lua [hunk 1]
```diff
@@ -1,143 +1,353 @@
...
 
+local show_formspec = function(name)
+	local page = current_pages[name]
+	local form = get_formspec(page, name)
+	minetest.show_formspec(name, "chest_of_everything:getitem", form)
+	return true
+end
+
+minetest.register_on_player_receive_fields(function(player, formname, fields)
+	if formname == "chest_of_everything:getitem" then
```
Comment (sfan5) [inline games/devtest/mods/chest_of_everything/init.lua:258]:
```suggestion
minetest.register_on_player_receive_fields(function(player, formname, fields)
	if formname ~= "chest_of_everything:getitem" then
		return
	end
	-- (rest of the code)
```

File: games/devtest/mods/chest_of_everything/init.lua [hunk 2]
```diff
@@ -1,143 +1,353 @@
...
-	"button[1,4;1,1;chest_of_everything_next;"..F(">").."]"..
-	"label[2,4;"..F("Page: "..page).."]"..
-	"listring[detached:everything;main]"..
+local detached_inventories = {}
+
+-- Per-player lists (indexed by player name)
+local current_pages = {} -- current page number
+local current_max_pages = {} -- current max. page number
+local current_searches = {} -- current search string
+local current_item_lists = {} -- currently active list of items
```
Comment (sfan5) [inline games/devtest/mods/chest_of_everything/init.lua:10]: unused, luacheck complained about this

**Comment by Wuzzy2** [comment]
@sfan5: Done.

**Review by sfan5 — APPROVED** [review]

**Comment by x2048** [comment]
Works and looks good.

**Pull request merged**